### PR TITLE
Map file extensions to language names for Rouge

### PR DESCRIPTION
**Pull request by ismaelga**
In an attempt to fix elixir code on the submission page ended up also fixing it for Perl, OCaml and C++.

### CODE REVIEW

**Comment by ismaelga** [comment]
Instead of this would it be better to use the language from submissions table instead of filename?

File: lib/app/helpers/markdown.rb [hunk 1]
```diff
@@ -4,9 +4,9 @@
 module ExercismWeb
   module Helpers
     module Markdown
-     def md(text, language = "")
-         language = language.split('.')[1] if language.include?('.')
-         #needs a test
```
Comment (ismaelga) [inline lib/app/helpers/markdown.rb]: oops. Didn't want to remove this comment

**Comment by kytrinyx** [comment]
Nice, thank you!

I'm curious about how the language is defined. We already have the language name on the problem, perhaps it is using the wrong bit of information to send to the markdown method?

Here's a list of the places it's used:

```plain
lib/app/helpers/markdown.rb:     def md(text, language = "")
lib/app/views/code/random.erb:        <%= md(code, filename) %>
lib/app/views/code/simple.erb:    <%= md(code, language) %>
lib/app/views/exercises/readme.erb:  <%= md(text) %>
lib/app/views/exercises/test_suite.erb:  <%= md(text, problem.track_id) %>
lib/app/views/site/conversations.erb:            <%= md(iteration.code, iteration.language) %>
lib/app/views/styleguide/_example.erb:<%= md(example, "html") %>
lib/app/views/submissions/current_submission.erb:      <%= md(code, filename) %>
test/app/helpers/markdown_test.rb:    output = helper.md(code, "ruby")
```

I notice that on `conversations` and `test_suite` it uses the `problem.track_id` (which is correct), and `iteration.language` (which in this case is an alias to the `track_id`).

We have the submission available to us, and the submission has a track_id -- let's use that.

What do you think?

**Comment by ismaelga** [comment]
Yes. I just realised it after the fix that we could use language or track_id instead of filename. But didn't knew if there were any reason for not using them.

Now it's using track_id when there is one.

**Comment by kytrinyx** [comment]
Superb, thank you! :heart:

**Pull request merged**